### ISSUE TITLE
How to create a backup using pgdump and REST-API

### ISSUE THREAD

**Issue by jorgesaenzpuchau** [CrunchyData/postgres-operator#879]
** Which example are you working with? **
Trying to create a backup of a single database in the cluster using pgdump, usign directly the REST API.

**What is the current behavior?**
Unable to create a backup using REST API and pgdump

**What is the expected behavior?**
Backup over a single database should be created.

**Other information** (e.g. detailed explanation, related issues,  etc)
When invoking the REST API with the following command
`curl --cacert $PGO_CA_CERT --key $PGO_CLIENT_KEY --cert $PGO_CA_CERT -u username:password -H "Content-Type:application/json" --insecure -X POST --data '{"ClientVersion":"4.0.0","Namespace":"eea-test","Selector":"name=poc-cluster-gis","BackupOpts":"--schema=datasets.dataset_1"}' https://10.103.130.3:8443/pgdumpbackup`

The process fails with the following message:
`{"Results":[],"Code":"error","Msg":"primary pod is not in Ready state"}`

Logs:
```
time="2019-07-23T09:37:16Z" level=debug msg="pgdumpservice.CreatepgDumpHandlerBackupHandler called"
time="2019-07-23T09:37:16Z" level=debug msg="Authentication Attempt CreateDump username=[username]"
time="2019-07-23T09:37:16Z" level=debug msg="BasicAuthzCheck pgoadmin CreateDump true"
time="2019-07-23T09:37:16Z" level=debug msg="Authentication Success"
time="2019-07-23T09:37:16Z" level=debug msg="GetNamespace username [username] ns [eea-test]"
time="2019-07-23T09:37:16Z" level=debug msg="WatchingNamespace [eea-test]"
time="2019-07-23T09:37:16Z" level=debug msg="CreatePgDump storage config... "
time="2019-07-23T09:37:16Z" level=debug msg="myselector is name=poc-cluster-gis"
time="2019-07-23T09:37:16Z" level=debug msg="create pgdump called for poc-cluster-gis"
time="2019-07-23T09:37:16Z" level=debug msg="jobs.batch \"poc-cluster-gis-pgdump-job\" not found"
time="2019-07-23T09:37:16Z" level=debug msg="task backup-poc-cluster-gis-pgdump not found "
time="2019-07-23T09:37:16Z" level=debug msg="pgdump pgtask backup-poc-cluster-gis-pgdump was not found so we will create it"
time="2019-07-23T09:37:16Z" level=error msg="primary pod is not in Ready state"
```

No job is created at any moment...

Cluster is created with the following command:
`curl --cacert $PGO_CA_CERT --key $PGO_CLIENT_KEY --cert $PGO_CA_CERT -u username:password -H "Content-Type:application/json" --insecure -X POST --data '{"ClientVersion":"4.0.0","Namespace":"eea-test","Name":"poc-cluster-gis","Series":1,"CCPImage":"crunchy-postgres-gis","PgpoolFlag":true,"ReplicaCount":1,"Password":"password"}' https://10.103.130.3:8443/clusters`

**Please tell us about your environment:**

* Operating System:
* Where is this running ( Local , Cloud Provider)
* Storage being used (NFS, Hostpath, Gluster, etc):
* Container Image Tag:
* PostgreSQL Version:
* Platform (Docker, Kubernetes, OpenShift): Kubernetes
* Platform Version:

If possible please run the following on the kubernetes or Openshit (oc) commands and provide the result:
 kubectl describe yourPodName
 kubectl describe pvc
 kubectl get nodes
 kubectl log yourPodName

**Comment by cahoonpwork** [comment]
This seems odd in the pgdump log: time="2019-07-23T09:37:16Z" level=error msg="primary pod is not in Ready state"

Can you confirm the create cluster worked and the primary is running.
Can you provide the following output:
 `kubectl get all -n eea-test` 
 `pgo show cluster poc-cluster-gis -n eea-test `
 `pgo test poc-cluster-gis -n eea-test`

**Comment by jorgesaenzpuchau** [comment]
Hi, thanbks for answering.

**kubectl get all -n eea-test**
```
NAME                                          READY   STATUS    RESTARTS   AGE
pod/poc-cluster-gis-7876848cb-ckbxj           1/1     Running   0          23h
pod/poc-cluster-gis-fvbk-6bc9c8788b-9vhnw     1/1     Running   0          23h
pod/poc-cluster-gis-pgpool-69458845cd-4nnkl   1/1     Running   0          23h

NAME                              TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)                                         AGE
service/poc-cluster-gis           ClusterIP   10.105.140.238   <none>        5432/TCP,9100/TCP,10000/TCP,2022/TCP,9187/TCP   23h
service/poc-cluster-gis-pgpool    ClusterIP   10.103.231.173   <none>        5432/TCP,9100/TCP,10000/TCP,2022/TCP,9187/TCP   23h
service/poc-cluster-gis-replica   ClusterIP   10.99.127.211    <none>        5432/TCP,9100/TCP,10000/TCP,2022/TCP,9187/TCP   23h
service/poc-external              NodePort    10.96.145.0      <none>        5432:32262/TCP                                  46h
service/poc-pgpool-external       NodePort    10.100.16.219    <none>        5432:31834/TCP                                  46h

NAME                                     READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/poc-cluster-gis          1/1     1            1           23h
deployment.apps/poc-cluster-gis-fvbk     1/1     1            1           23h
deployment.apps/poc-cluster-gis-pgpool   1/1     1            1           23h

NAME                                                DESIRED   CURRENT   READY   AGE
replicaset.apps/poc-cluster-gis-7876848cb           1         1         1       23h
replicaset.apps/poc-cluster-gis-fvbk-6bc9c8788b     1         1         1       23h
replicaset.apps/poc-cluster-gis-pgpool-69458845cd   1         1         1       23h
```

**pgo show cluster poc-cluster-gis -n eea-test**
Executed as 
`curl --cacert $PGO_CA_CERT --key $PGO_CLIENT_KEY --cert $PGO_CA_CERT -u username:password -H "Content-Type:application/json" --insecure -X POST --data '{"ClientVersion":"4.0.0","Namespace":"eea-test","Clustername":"poc-cluster-gis"}' https://10.103.130.3:8443/showclusters`

See file 
[show_cluster.txt](https://github.com/CrunchyData/postgres-operator/files/3425920/show_cluster.txt)


**pgo test poc-cluster-gis -n eea-test**
Executed as 
`curl --cacert $PGO_CA_CERT --key $PGO_CLIENT_KEY --cert $PGO_CA_CERT -u username:password -H "Content-Type:application/json" --insecure -X POST --data '{"ClientVersion":"4.0.0","Namespace":"eea-test","Clustername":"poc-cluster-gis"}' https://10.103.130.3:8443/showclusters`

See file
[test_cluster.txt](https://github.com/CrunchyData/postgres-operator/files/3425933/test_cluster.txt)

Regards
JSP

**Comment by cahoonpwork** [comment]
@jorgesaenzpuchau We reproduced your error and are looking into it.

**Comment by cahoonpwork** [comment]
Issue is fixed on the develop branch and will be in the next release. https://github.com/CrunchyData/postgres-operator/pull/894

**Comment by jkatz** [comment]
This was released in October as part of v4.1.0
